### ISSUE TITLE
could this be called from the command line?

### ISSUE THREAD

**Issue by Kreijstal** [nbesli/pdf-merger-js#72]
it'd be nice, because it only depends on node

**Comment by mojoaxel** [comment]
@Kreijstal Sorry i've overlook this! This is a very nice idea!

Do you have any idea how a interface could look? Should it only be possible to merg two documents or multibles at the same time? What about page number?

Let's discuss this idea here any further!

**Comment by mojoaxel** [comment]
Here is a proposal:

```sh
pdf-merger-js --out=merged.pdf input1.pdf[1,2] input2.pdf input3.pdf[3-6]
```

So this would merge 
page 1 and 2 of input1.pdf
with input2.pdf
and with pages 3 to 6 of input3.pdf
and save everything to merged.pdf

**Comment by mojoaxel** [comment]
For very basic merging you could use [pdf-merger-cli](https://github.com/zorji/pdf-merger-cli).

**Comment by Kreijstal** [comment]
> For very basic merging you could use [pdf-merger-cli](https://github.com/zorji/pdf-merger-cli).

could we merge these with the original? very cool by the way

**Comment by mojoaxel** [comment]
Here is another proposal I like. This would also keep urls valid:

`pdf-merger-js --out=merged.pdf input1.pdf#1,2 input2.pdf input3.pdf#3-6`

**Comment by sabine33** [comment]
Please look at this package, It does work exactly as @mojoaxel wanted. 

[](https://github.com/sabine33/pdf-merger )

**Comment by sabine33** [comment]
Pleas have a look at it.
https://www.npmjs.com/package/merge-the-pdfs

https://github.com/sabine33?tab=repositories

**Comment by mojoaxel** [comment]
Here is a overview about existing solutions

### [merge-the-pdfs](https://github.com/sabine33/pdf-merger)

```
$ merge-the-pdfs --folder ~/Downloads/pdfs --output filename.pdf --files abc.pdf#1,2 apple.pdf#3,4
$ merge-the-pdfs --output filename.pdf --files abc.pdf#1,2 apple.pdf#3,4
```

### [pdf-merger-cli](https://github.com/zorji/pdf-merger-cli)

```
$ npx pdf-merger-cli out.pdf input1.pdf input2.pdf
$ npx pdf-merger-cli out.pdf input*.pdf
```

### [merge-pdf-cli](https://github.com/g1patil/merge-pdf-cli)

```
$ merge-pdf inputfile-1.pdf inputfile-2.pdf
$ Successfully merged pdfs.  Output : merged-file.pdf
```

**Comment by mojoaxel** [comment]
Looking at the existing solutions I would propose the following cli:

#### Simple inputs
```
pdf-merger-js --output merged.pdf input1.pdf input2.pdf
```

#### Select pages
```
pdf-merger-js --output merged.pdf input1.pdf#1-3 input2.pdf#3,4,6
```

#### Glob inputs
```
pdf-merger-js --output merged.pdf ./**/input*.pdf
```

#### URL inputs
```
pdf-merger-js --output merged.pdf https://example.org/input1.pdf#1-2 https://example.org/input2.pdf#3-6
```

**Comment by mojoaxel** [comment]
This has been implemented with #130 :tada:

**Comment by mojoaxel** [comment]
The command-line-interface is available since [v5.1.1](https://github.com/nbesli/pdf-merger-js/releases/tag/v5.1.1) :rocket:
